File: backend/src/main/java/acalidonio/bodegamovilbackend/domain/entities/Product.java
package acalidonio.bodegamovilbackend.domain.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    private String sku;
    
    private String name;
    private String description;
    
    @Enumerated(EnumType.STRING)
    private ProductCategory category;
    
    @Enumerated(EnumType.STRING)
    private StockStatus status;
    private String location;
    private int stock;
    private Date lastAudit;
    
    // Technical specs
    private String innerDiameter;
    private String outerDiameter;
    private String width;
    private String weight;
    private String material;
    
    @Builder.Default
    @Column(columnDefinition = "boolean default true")
    private boolean active = true;
    
    private String imageUrl;
}

File: app/src/main/java/com/acalidonio/bodegamovil/model/Product.kt
package com.acalidonio.bodegamovil.model

data class Product(
    val sku: String,
    val name: String,
    val location: String,
    val stock: Int,
    val status: StockStatus,
    val lastAudit: String,
    // Technical specs
    val innerDiameter: String? = null,
    val outerDiameter: String? = null,
    val width: String? = null,
    val weight: String? = null,
    val material: String? = null
)

enum class StockStatus {
    AVAILABLE,
    LOW_STOCK,
    OUT_OF_STOCK
}
